### PR TITLE
fix switch over candidate retrieving

### PR DESCRIPTION
**Pull request by fahedouch**
fixe [issue](https://github.com/zalando/postgres-operator/issues/2751).

When asynchronous mode is enabled and no candidates are found, no retry is currently triggered to fetch a switchover candidate from the Patroni API. This PR enables retry for asynchronous mode, similar to [synchronous mode](https://github.com/zalando/postgres-operator/blob/master/pkg/cluster/pod.go#L485-L486).

### CODE REVIEW

**Comment by Falcondevils** [comment]
Need this fix. I am facing the same issue

File: pkg/cluster/pod.go [hunk 1]
```diff
@@ -489,6 +492,13 @@ func (c *Cluster) getSwitchoverCandidate(master *v1.Pod) (spec.NamespacedName, e
 				return false, nil
 			}
 
+			// if asynchronous mode is enabled and no candidates was found
+			// return false for retry - cannot switchover with no replicas candidate
+			if !c.Spec.Patroni.SynchronousMode && len(candidates) == 0 {
+				c.logger.Warnf("no replicas candidate found - retrying fetching cluster members")
```
Comment (FxKu) [inline pkg/cluster/pod.go:498]:
```suggestion
				c.logger.Warnf("no replica candidate found - retrying fetching cluster members")
```

File: pkg/cluster/pod.go [hunk 2]
```diff
@@ -489,6 +492,13 @@ func (c *Cluster) getSwitchoverCandidate(master *v1.Pod) (spec.NamespacedName, e
 				return false, nil
 			}
 
+			// if asynchronous mode is enabled and no candidates was found
+			// return false for retry - cannot switchover with no replicas candidate
```
Comment (FxKu) [inline pkg/cluster/pod.go:496]:
```suggestion
			// retry also in asynchronous mode when no replica candidate was found
```

**Comment by FxKu** [comment]
:+1:

**Comment by FxKu** [comment]
Thanks @fahedouch for your contribution and improving the operator experience :smiley:

**Pull request merged**